nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/main/java/org/apache/nifi/processors/solr/FetchSolr.java

```diff
@@ -0,0 +1,401 @@
+            .description("Results of faceted search").build();
+    public static final Relationship STATS = new Relationship.Builder().name("stats")
+            .description("Stats about Solr index").build();
+    public static final Relationship ORIGINAL = new Relationship.Builder().name("original")
+            .description("Original flowfile").build();
+    public static final Relationship FAILURE = new Relationship.Builder().name("failure")
+            .description("Failure relationship").build();
+
+    private Set<Relationship> relationships;
+    private List<PropertyDescriptor> descriptors;
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return this.descriptors;
+    }
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        super.init(context);
+
+        final List<PropertyDescriptor> descriptors = new ArrayList<>();
+        descriptors.add(SOLR_TYPE);
+        descriptors.add(SOLR_LOCATION);
+        descriptors.add(COLLECTION);
+        descriptors.add(SOLR_QUERY_STRING);
+        descriptors.add(RETURN_TYPE);
+        descriptors.add(RECORD_WRITER);
+        descriptors.add(REQUEST_HANDLER);
+        descriptors.add(JAAS_CLIENT_APP_NAME);
+        descriptors.add(BASIC_USERNAME);
+        descriptors.add(BASIC_PASSWORD);
+        descriptors.add(SSL_CONTEXT_SERVICE);
+        descriptors.add(SOLR_SOCKET_TIMEOUT);
+        descriptors.add(SOLR_CONNECTION_TIMEOUT);
+        descriptors.add(SOLR_MAX_CONNECTIONS);
+        descriptors.add(SOLR_MAX_CONNECTIONS_PER_HOST);
+        descriptors.add(ZK_CLIENT_TIMEOUT);
+        descriptors.add(ZK_CONNECTION_TIMEOUT);
+        this.descriptors = Collections.unmodifiableList(descriptors);
+
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(FAILURE);
+        relationships.add(RESULTS);
+        relationships.add(FACETS);
+        relationships.add(STATS);
+        relationships.add(ORIGINAL);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    public static final Set<String> SUPPORTED_SEARCH_COMPONENTS = new HashSet<String>();
+    static {
+        SUPPORTED_SEARCH_COMPONENTS.addAll(Arrays.asList(StatsParams.STATS, FacetParams.FACET));
+    }
+    public static final Set<String> SEARCH_COMPONENTS_ON = new HashSet<String>();
+    static {
+        SEARCH_COMPONENTS_ON.addAll(Arrays.asList("true", "on", "yes"));
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        final ComponentLog logger = getLogger();
+
+        FlowFile flowFileOriginal = session.get();
+
+        if (flowFileOriginal == null) {
+            if (context.hasNonLoopConnection())
+                return;
+            flowFileOriginal = session.create();

```

Comment:
I didn't see you writing anything to this flowfile. It looks like what you're trying to do is bind together the provenance of all of the query response-related flowfiles. If that's the case, you should strongly consider writing the query to the flowfile so that users who have downstream components that examine the query can access it (which they might if the query is dynamic from the input).